File: src/Scripting/Lua/patterns/broker.lua
local broker = { handlers = {} }
function broker:register(topic, fn) self.handlers[topic] = fn end
function broker:request(topic, payload) return self.handlers[topic](payload) end
broker:register("square", function(x) return x*x end)
assert(broker:request("square", 5) == 25)
return true

File: src/Scripting/Ruby/patterns/broker.rb
# frozen_string_literal: true

module RubyPatterns
  module Broker
    module_function

    def run
      handlers = { price: ->(_sku) { 9 } }
      request = ->(topic, payload) { handlers.fetch(topic).call(payload) }
      raise 'Broker failed' unless request.call(:price, 'A') == 9
      true
    end
  end
end

RubyPatterns::Broker.run if $PROGRAM_NAME == __FILE__
